src/TestListeners/Subscribers/AddErrorSubscriber.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Yoast\PHPUnitPolyfills\TestListeners\Subscribers;
+
+use PHPUnit\Event\Test\Errored;
+use PHPUnit\Event\Test\ErroredSubscriber;
+
+/**
+ * Event subscriber.
+ *
+ * @since 2.0.0
+ */
+final class AddErrorSubscriber implements ErroredSubscriber {
+
+	/**
+	 * Subscriber constructor.
+	 *
+	 * @param object $handler Instance of the class which functions as the "test listener".
+	 */
+	public function __construct( private readonly object $handler ) {}
```

Comment:
@jrfnl 

Do you think it would make sense to use `PHPUnit\Framework\TestListener` as type declaration here (and in other subscribers)?

If the interface does not exist, you could declare it in the polyfill. What do you think?

Reply:
@jrfnl 

Alternatively, to have a specific type declaration here, what do you think about extracting a class that works as handler - the handler itself could compose an implementation of `PHPUnit\Framework\TestListener`. 

See https://github.com/Yoast/PHPUnit-Polyfills/pull/115!